aws-replicator/aws_replicator/client/http2_server.py

```diff
@@ -0,0 +1,319 @@
+# TODO: currently this is only used for the auth_proxy. replace at some point with the more modern gateway
+#  server
+import asyncio
+import collections.abc
+import logging
+import os
+import ssl
+import threading
+import traceback
+from typing import Callable, List, Tuple
+
+import h11
+from hypercorn import utils as hypercorn_utils
+from hypercorn.asyncio import serve, tcp_server
+from hypercorn.config import Config
+from hypercorn.events import Closed
+from hypercorn.protocol import http_stream
```

Comment:
Would be great to add `hypercorn` to the list of dependencies of this extension.
While `quart` is explicitly set in the dependencies of this extension, `hypercorn` and its transitive dependency `h11` are not. This is not yet a problem for removing `quart` from the list of dependencies in `localstack-core`, but since `hypercorn` is not the default gateway implementation anymore it is likely to be removed sometime soon as well.